tests/e2e/pageobjects/ide/Ide.ts

```diff
@@ -41,6 +41,57 @@ export class Ide {
     }
 
+    async waitDialogShell(timeout: number, openLink: boolean, closeDialog: boolean) {
+        const dialogShellLocator: By = By.id('theia-dialog-shell');
+        const dialogBlockLocator: By = By.xpath('.//div[@class="dialogBlock"]');
+        const dialogBlockTitleLocator: By = By.xpath('.//div[@class="dialogTitle"]/div');
+        const dialogBlockTitleCloseButtonLocator: By = By.xpath('.//div[@class="dialogTitle"]/i[@class="fa fa-times closeButton"]');
+        const dialogBlockNotificationTextLocator: By = By.xpath('.//div[@class="dialogContent"]/div[@class="modal-Notification"]/div[@class="text"]/span');
+        const dialogBlockControlButtonMainLocator: By = By.xpath('.//div[@class="dialogControl"]/button[@class="theia-button main"]');
+        const dialogBlockControlButtonSecondaryLocator: By = By.xpath('.//div[@class="dialogControl"]/button[@class="theia-button secondary"]');
+        Logger.debug(`Ide.waitDialogShell timeout:"${timeout}"`);
+
+        Logger.debug(`Ide.waitDialogShell waiting for dialog shell to become visible.`);
+        await this.driverHelper.waitVisibility(dialogShellLocator, timeout);
+
+        Logger.debug(`Ide.waitDialogShell get dialog shell.`);
+        const dialogShell: WebElementPromise = (await this.driverHelper.getDriver()).findElement(dialogShellLocator);
+        Logger.debug(`Ide.waitDialogShell finding dialogBlock child element.`);
+        const dialogBlock: WebElementPromise = (await dialogShell).findElement(dialogBlockLocator);
+
+        Logger.debug(`Ide.waitDialogShell getting dialogBlock title.`);
+        const dialogBlockTitle: WebElementPromise = (await dialogBlock).findElement(dialogBlockTitleLocator);
+        Logger.debug(`Ide.waitDialogShell getting dialogBlock title close button.`);
+        const dialogBlockTitleCloseButton: WebElementPromise = (await dialogBlock).findElement(dialogBlockTitleCloseButtonLocator);
+        Logger.debug(`Ide.waitDialogShell getting dialogBlock content text.`);
+        const dialogBlockNotificationText: WebElementPromise = (await dialogBlock).findElement(dialogBlockNotificationTextLocator);
+        Logger.debug(`Ide.waitDialogShell getting dialogBlock buttons.`);
+        const dialogBlockControlButtonMain: WebElementPromise = (await dialogBlock).findElement(dialogBlockControlButtonMainLocator);
+        const dialogBlockControlButtonSecondary: WebElementPromise = (await dialogBlock).findElement(dialogBlockControlButtonSecondaryLocator);
+        const primaryButtonText: string = (await (await dialogBlockControlButtonMain).getText()).toString();
+        const secondaryButtonText: string = (await (await dialogBlockControlButtonSecondary).getText()).toString();
+
+        Logger.debug(`Ide.waitDialogShell parsed dialog[ title:"${(await (await dialogBlockTitle).getText()).toString()}" notification_text:"${(await (await dialogBlockNotificationText).getText()).toString()}" ]`);
+
+        if (openLink) {
```

Comment:
It seems to me that `waitDialogShell` should just wait until a shell is visible. I would separate this logic to some parameterized method that can be used dynamically. Something like this:
```
async clickDialogButton(text: string) {
          const primaryButtonText: string = (await (await dialogBlockControlButtonMain).getText()).toString();
          const secondaryButtonText: string = (await (await dialogBlockControlButtonSecondary).getText()).toString();

           if (primaryButtonText.match(text)) {
                (await dialogBlockControlButtonMain).click();
            } else if (secondaryButtonText.match(text)) {
                (await dialogBlockControlButtonSecondary).click();
            } 
}

async closeDialog() {
            await dialogBlockTitleCloseButton).click();
}
```

Maybe there would be some better dynamic way how to iterate through all buttons, because maybe the dialog can have more than 2 buttons.

Reply:
I've rewritten the logic. In my experience I haven't seen it used in any other way, but I would be happy to make it universal if the case appears.

For now I've deleted the logic that searches for the button text as `main` always seems to be 'Open link' and `secondary` seems to be `'close'`

Reply:
Not relevant anymore, using DialogWindow methods now.